src/main/java/seedu/duke/Api.java

```diff
@@ -0,0 +1,42 @@
+package seedu.duke;
+
+import com.google.gson.Gson;
+import netscape.javascript.JSObject;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URI;
+import java.net.URL;
+import java.net.URLConnection;
+import java.net.http.HttpClient;
+import java.util.List;
+
+public class Api {
+
+    String urlString = "https://api.nusmods.com/v2/2020-2021/moduleList.json";
+
+    public Api() {
+    }
+
+    public Module[] getAllModules() throws IOException {
+        URL url = new URL(urlString);
+        URLConnection conn = url.openConnection();
+        InputStream is = conn.getInputStream();
+        BufferedReader bR = new BufferedReader(  new InputStreamReader(is));
+        String line = "";
+
+        StringBuilder responseStrBuilder = new StringBuilder();
+        while((line =  bR.readLine()) != null){
```

Comment:
> WhitespaceAround: 'while' is not followed by whitespace. Empty blocks may only be represented as {} when not part of a multi-block statement (4.1.3) [WhitespaceAround].

 I think the while need to have a space after it as indicated by the error. But it seem weird to have a space between there. Can try adding a space to see if the error will be solve?